lib/krb5/fcache.c

```diff
@@ -1162,22 +1169,225 @@ fcc_get_version(krb5_context context,
     return FCACHE(id)->version;
 }
 
+static const char *
+my_basename(const char *fn)
+{
+    const char *base;
+
+    if (strncmp(fn, "FILE:", sizeof("FILE:") - 1))
+        return "";
+    fn += sizeof("FILE:") - 1;
+    for (base = fn + strlen(fn); base >= fn; base--) {
+#ifdef WIN32
+        if (*base == '/' || *base == '\\')
+            break;
+#else
+        if (*base == '/')
+            break;
+#endif
+    }
+    return base + 1;
+}
+
+static int
+matchfile(const char *fn, const char *base, int match_sub_only)
+{
+    size_t baselen;
+
+    if (strncmp(base, "FILE:", sizeof("FILE:") - 1) == 0)
+        base += sizeof("FILE:") - 1;
+
+    baselen = strlen(base);
+    return strncmp(fn, base, baselen) == 0 &&
+        ((!match_sub_only && fn[baselen] == '\0') ||
+         (fn[baselen] == ':' && fn[baselen + 1] != '\0'));
```

Comment:
Don't know what's going on here. Please add commentary what `match_sub_only` is supposed to mean.

Reply:
Commentary added.